### PR TITLE
Add RTS interfaces

### PR DESCRIPTION
**Pull request by disha-wani**
**What this PR does / why we need it**: This PR adds  RTS interfaces.

**Special notes for your reviewer**: NONE

**Release note**NONE

### CODE REVIEW

**Comment by Savasw** [comment]
@niuzhenguo Can you please help update if the coverall result would be blocker for merge. Also suggest what steps could possibly help resolve it.

**Comment by niuzhenguo** [comment]
@Savasw That's because tests coverage decreased with the change, you can check the file list above to find which parts missing coverage. I'm not sure if it's a blocker for merge now but we do not care much about that before :P. @freesky-edward can you please help to confirm?

File: openstack/rts/v1/stacks/utils.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+	fileMaps map[string]string
+	// baseURL represents the location of the template or environment file.
+	baseURL string
+	// client is an interface which allows TE to fetch contents from URLS
+	client Client
+}
+
+// Fetch fetches the contents of a TE from its URL. Once a TE structure has a
+// URL, call the fetch method to fetch the contents.
+func (t *TE) Fetch() error {
```
Comment (freesky-edward) [inline openstack/rts/v1/stacks/utils.go]: if this is an internal method, I would prefer to name it starting with lowercase letter.

**Comment by Savasw** [comment]
@freesky-edward As discussed, for RTS Stack, we have followed code architecture from openstack gophercloud where TE is abstract for both environment and template as there are lot of common features that work for both. Also Fetch function is not used externally at the moment but could be useful for end users separately to fetch content from URL.

**Comment by freesky-edward** [comment]
@Savasw thanks for your response, LGTM

**Pull request merged**